src/AggregateRoots/AggregateRoot.php

```diff
@@ -129,6 +157,42 @@ public function persist(): static
         return $this;
     }
 
+    public function persistConcurrently(int $tries = -1): static
+    {
+        $recordedEvents = $this->recordedEvents;
+        $concurrencyChecks = $this->concurrencyChecks;
+
+        try {
+            $storedEvents = $this->persistWithoutApplyingToEventHandlers();
+        } catch (CouldNotPersistAggregate $exception) {
```

Comment:
One concern: are there any ways `CouldNotPersistAggregate` is thrown in cases where you can't recover from retrying? We should be sure you can't end up in an infinite loop.

Reply:
Hmm I can't think of any other places… Do you think we should not allow `-1` for tries and have a hard limit of `100`?

Reply:
It might be a good failsafe? 100 seems good enough, I cannot imagine anyone wanting to retry a 100 times, and if they want to, they can still overwrite the value. But better to be safe than sorry, I think?